dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -1007,4 +1011,151 @@ public void resetSecrets(final User user)
        appsCache.clearCache();
     }
 
+    /**
+     * {@inheritDoc}
+     * @param key
+     * @param paramAppKeysBySite
+     * @return
+     */
+    public Path exportSecrets(final Key key, final boolean exportAll,
+            final Map<String, Set<String>> paramAppKeysBySite, final User user)
+            throws DotDataException, DotSecurityException, IOException {
+
+        if(!user.isAdmin()){
+            throw new DotSecurityException("Only Admins are allowed to perform an export operation.");
+        }
+
+        final AppsSecretsImportExport exportedSecrets;
+        if (exportAll) {
+            exportedSecrets = collectSecretsForExport(appKeysByHost(), user);
+        } else {
+            exportedSecrets = collectSecretsForExport(paramAppKeysBySite, user);
+        }
+
+        Logger.info(AppsAPIImpl.class,""+exportedSecrets);
```

Comment:
Codacy found an issue: [Do not add empty strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5931733)